src/layouts/EditHomepage.jsx

```diff
@@ -238,13 +239,39 @@ export default class EditHomepage extends Component {
           const sectionIndex = parseInt(idArray[1], RADIX_PARSE_INT);
           const sectionType = idArray[2]; // e.g. "hero" or "infobar" or "resources"
           const field = idArray[3]; // e.g. "title" or "subtitle"
-
           sections[sectionIndex][sectionType][field] = value;
 
+          let newSectionError
+
+          // Set special error message if hero button has text but hero url is empty
+          // This needs to be done separately because it relies on the state of another field
+          if (
+            field === 'url' && !value && this.state.frontMatter.sections[sectionIndex][sectionType].button
+            && !(!this.state.frontMatter.sections[sectionIndex][sectionType].button && !this.state.frontMatter.sections[sectionIndex][sectionType].url)
+          ) {
+            const errorMessage = 'Please specify a URL for your button'
+            newSectionError = _.cloneDeep(errors.sections[sectionIndex])
+            newSectionError[sectionType][field] = errorMessage
+          } else if (
+            field === 'button' && !this.state.frontMatter.sections[sectionIndex][sectionType].url
+            && !(!this.state.frontMatter.sections[sectionIndex][sectionType].button && !this.state.frontMatter.sections[sectionIndex][sectionType].url)
```

Comment:
Nit: the double negatives here are a bit confusing, could we change them to (this.state.frontMatter.sections[sectionIndex][sectionType].button || this.state.frontMatter.sections[sectionIndex][sectionType].url) instead?

Reply:
fixed in [267650b](https://github.com/isomerpages/isomercms-frontend/pull/230/commits/267650b7fac8e2dbed211cd544dfbd3b76974a3a)